consensus/istanbul/backend/engine_test.go

```diff
@@ -354,29 +99,37 @@ func TestSealStopChannel(t *testing.T) {
 // TestSealCommittedOtherHash checks that when Seal() ask for a commit, if we send a
 // different block hash, it will abort
 func TestSealCommittedOtherHash(t *testing.T) {
-	chain, engine := newBlockChain(4, true)
-	block := makeBlockWithoutSeal(chain, engine, chain.Genesis())
+	for numValidators := 1; numValidators < 5; numValidators++ {
+		testSealCommittedOtherHash(t, numValidators)
+	}
+}
 
+func testSealCommittedOtherHash(t *testing.T, numValidators int) {
+	chain, engine := newBlockChain(numValidators, true)
+	block := makeBlockWithoutSeal(chain, engine, chain.Genesis())
 	// create a second block which will have a different hash
-	otherBlock := makeBlockWithoutSeal(chain, engine, chain.Genesis())
-	eventSub := engine.EventMux().Subscribe(istanbul.RequestEvent{})
-	eventLoop := func() {
-		ev := <-eventSub.Chan()
-		_, ok := ev.Data.(istanbul.RequestEvent)
-		if !ok {
-			t.Errorf("unexpected event comes: %v", reflect.TypeOf(ev.Data))
-		}
-		engine.Commit(otherBlock, types.IstanbulAggregatedSeal{}, types.IstanbulEpochValidatorSetSeal{})
-		eventSub.Unsubscribe()
+	h := block.Header()
+	h.ParentHash = block.Hash()
+	otherBlock := types.NewBlock(h, nil, nil, nil)
+	if block.Hash() == otherBlock.Hash() {
+		t.Fatalf("did not create different blocks")
 	}
-	go eventLoop()
+
 	results := make(chan *types.Block)
+	engine.Seal(chain, block, results, nil)
+
+	// in the single validator case, the result will instantly be processed
+	// so we should discard it and check that the Commit will not do anything
+	if numValidators == 1 {
+		<-results
+	}
+	// this commit should _NOT_ push a new message to the queue
+	engine.Commit(otherBlock, types.IstanbulAggregatedSeal{}, types.IstanbulEpochValidatorSetSeal{})
 
-	go func() { engine.Seal(chain, block, results, nil) }()
 	select {
-	case <-results:
-		t.Error("seal should not be completed")
-	case <-time.After(2 * time.Second):
+	case res := <-results:
+		t.Fatal("seal should not be completed", res)
+	case <-time.After(100 * time.Millisecond):
```

Comment:
Why can this be so much faster now?

Reply:
This was leftover from the initial instanbul implementation, it could've been smaller anyway